packages/waku/src/router/define-router.ts

```diff
@@ -379,6 +379,9 @@ export function new_defineRouter(fns: {
       pathname,
       pathStatus.isStatic ? {} : { query, skip },
     );
+    if (entries === null) {
+      return null;
+    }
```

Comment:
When is this required?

Reply:
At least for now, this is what is returned when a path is requested that matches no components: https://github.com/dai-shi/waku/pull/963/files#diff-e2495518ca4a32ae11025d1ecca9cb482f9cba4f15c1d8eb8d99bc50cb5f2d13R702

Reply:
Yeah. I wonder if this would be a hint to revisit new_defineRouter API.
Don't you feel the responsibility between `getPathConfig` and `renderRoute` is a bit vague?

Reply:
it does seem like there's overlapping responsibility from my point of view for now.

The weird part to me is that `getPathConfig` returns all the path specs, then `new_defineRouter` uses each of these to then request each path one at a time. 

The simplest solution that would solve this that I can think of is if createPages had a single function for handling the current split responsibilities of `renderRoute` and `getPathConfig`. So it would return a `component` per each path spec for example. 

Perhaps there are problems or limitations on control with this that you can think of maybe? 

A benefit of reducing to a single param for `new_defineRouter` would be that the skip list could be entirely managed from inside `new_defineRouter`.

As I write this out, I am realizing that this idea of consolidation could come at the cost of maybe taking away cache capabilities that `new_createPages` could later implement, so maybe this helps generate a new idea for you?

Reply:
@dai-shi Should I continue rewriting createPages for the current version of new_defineRouter or should we resolve this thread before continuing considering the possibility of more breaking new_defineRouter changes?

Reply:
new_defineEntries will be more low level api, so new_defineRouter should have less responsibility and delegate to createPages.
new_defineRouter should only responsible for waku/router/client capability, and it should be as thin as possible.

I'm not sure what's your latest proposition, but if it follow the above principle, it's considerable.